apps/kyb-app/src/components/organisms/VersionResolver/VersionResolver.tsx

```diff
@@ -0,0 +1,33 @@
+import { LoadingScreen } from '@/pages/CollectionFlow/components/atoms/LoadingScreen';
+import { AnyChildren } from '@ballerine/ui';
+import { FunctionComponent, useEffect, useState } from 'react';
+
+interface IVersionResolverProps {
+  version: number;
+  children: AnyChildren;
+}
+
+export const VersionResolver: FunctionComponent<IVersionResolverProps> = ({
+  version,
+  children,
+}) => {
+  const [isResolved, setResolved] = useState(false);
+
+  useEffect(() => {
+    if (!version) return;
```

Comment:
Try to give to a developer working on this some reflection on edge cases, as I understand if there is no version he will just see loader and nothing in the console